src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,651 @@
+/-
+Copyright (c) 2021 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import measure_theory.vector_measure
+import order.symm_diff
+
+/-!
+# Unsigned Hahn decomposition
+
+This file defines positive and negative sets with respect to a vector measure and
+prove the Hahn decomposition (signed version).
+
+## Main definitions
+
+* `vector_measure.positive` : a set `i` is positive with respect to a vector measure if for all
+  measurable sets `j` contained in `i`, the measure of `j` is non-negative.
+* `vector_measure.negative` : a set `i` is negative with respect to a vector measure if for all
+  measurable sets `j` contained in `i`, the measure of `j` is non-positive.
+
+## Main results
+
+* `signed_measure.exists_disjoint_positive_negative_union_eq` : the Hahn decomposition theorem.
+* `signed_measure.exists_negative_set` : A measurable set of negative measure contains at least
+  one negative subset.
+
+## Tags
+
+positive, negative, Hahn decomposition
+-/
+
+noncomputable theory
+open_locale classical big_operators nnreal ennreal
+
+section lemmas
+-- ↓ I will move the content of this section to the approprate files after the reviews so we don't
+-- need to compile all of mathlib everytime.
+
+variables {α : Type*}
+
+open set filter
+
+lemma set.union_inter_diff_eq {a b c : set α} (habc : a ⊆ b ∪ c) :
+  a ∩ b ∪ a ∩ c \ (a ∩ b) = a :=
+by rwa [union_diff_self, ← inter_union_distrib_left, inter_eq_left_iff_subset]
```

Comment:
This generalizes to:
```
lemma sup_inf_diff_eq {α} {a b c : α} [boolean_algebra α] (habc : a ≤ b ⊔ c) :
  a ⊓ b ⊔ a ⊓ c \ (a ⊓ b) = a :=
by rwa [sup_sdiff_self_right, ←inf_sup_left, inf_eq_left]

lemma set.union_inter_diff_eq {α} {a b c : set α} (habc : a ⊆ b ∪ c) :
  a ∩ b ∪ a ∩ c \ (a ∩ b) = a := sup_inf_diff_eq
```
but I'm not convinced either are lemma worthy.

Reply:
I'll get rid of them by including them in the proof in that case